docs/hive-integration/kube-api-getting-started.md

```diff
@@ -0,0 +1,493 @@
+time="2021-06-29T15:45:33Z" level=debug msg="Authenticating cluster 02a89bb9-6141-4d14-a82e-42f254217502 JWT" func="github.com/openshift/assisted-service/pkg/auth.(*LocalAuthenticator).AuthAgentAuth" file="/go/src/github.com/openshift/origin/pkg/auth/local_authenticator.go:82" pkg=auth
+time="2021-06-29T15:45:33Z" level=debug msg="Pushing cluster event single-node demo-worker4" func="github.com/openshift/assisted-service/internal/controller/controllers.(*controllerEventsWrapper).AddEvent" file="/go/src/github.com/openshift/origin/internal/controller/controllers/controller_event_wrapper.go:36"
+time="2021-06-29T15:45:33Z" level=info msg="ClusterDeployment Reconcile started" func="github.com/openshift/assisted-service/internal/controller/controllers.(*ClusterDeploymentsReconciler).Reconcile" file="/go/src/github.com/openshift/origin/internal/controller/controllers/clusterdeployments_controller.go:115" cluster_deployment=single-node cluster_deployment_namespace=demo-worker4 go-id=684 request_id=9d4b6bae-b3a2-420f-ad4b-12bb4fa3d5da
+time="2021-06-29T15:45:33Z" level=info msg="ClusterDeployment Reconcile ended" func="github.com/openshift/assisted-service/internal/controller/controllers.(*ClusterDeploymentsReconciler).Reconcile.func1" file="/go/src/github.com/openshift/origin/internal/controller/controllers/clusterdeployments_controller.go:112" agent_cluster_install=test-agent-cluster-install agent_cluster_install_namespace=demo-worker4 cluster_deployment=single-node cluster_deployment_namespace=demo-worker4 go-id=684 request_id=9d4b6bae-b3a2-420f-ad4b-12bb4fa3d5da
+```
+
+###### Inspect Assisted Service Cluster Events
+Get the full event list here:
+```bash
+curl -s -k $(kubectl -n demo-worker4 get agentclusterinstalls.extensions.hive.openshift.io test-agent-cluster-install -o=jsonpath="{.status.debugInfo.eventsURL}")  | jq "."
+```
+
+Expect to see an event that indicate the image download started:
+```json
+  {
+    "cluster_id": "02a89bb9-6141-4d14-a82e-42f254217502",
+    "event_time": "2021-06-29T15:45:33.489Z",
+    "message": "Started image download (image type is \"minimal-iso\")",
+    "severity": "info"
+  }
+```
+### Agent Discovery and Installation
+* When the host boots from the discovery image, it gets registered to assisted-service. At first it will look like this:
+```bash
+kubectl -n demo-worker4 get agents
+```
+```bash
+NAME                                   CLUSTER        APPROVED
+07e80ea9-200c-4f82-aff4-4932acb773d4   test-cluster   false
+```
+* BMAC will automatically approve the agent, which will start the installation.
```

Comment:
Assuming this doc is not specific to bare metal or ZTP, I would add here "If you used BMAC, it will..." Then explain how to approve it otherwise.

Reply:
As you proposed, I made a different advanced section for this.
At the basic section (boot it yourself), I demonstrate how to manually approve the agent.